### PR TITLE
bluespace river away site update

### PR DESCRIPTION
**Pull request by Ilysen**
## About the Pull Request

The bluespace river's super old, and I wanted to make some adjustments to it. Here's the rundown, with user-facing changes at the top and more backend-related ones at the bottom.
* The catwalk in the bluespace river is now 2-wide, and has guard rails.
* Above-ground and underground sections of the bluespace river are now connected with stairs, instead of a ladder. This lets objects and people be pulled up and down.
* Remapped the entrance section of the bluespace river to be a natural-looking hollow instead of a square. Some furniture and lockers have been added.
* Rewrote almost all of the text on the away site, including notes and the flavor text of items found there.
* Bluespace river tiles are now floors, instead of walls. This means that you can see it through the catwalks and throw items onto it! Flying mobs are also able to traverse the river, to a point - going too far will show a message and prevent them from continuing further.
* Impassable rock turfs now give a message when attempting to mine them, to make it obvious that you can't get past them.
* Bluespace river tiles, and the aesthetic crystal clusters scattered around, can be hit with an anomaly scanner for a big fluff readout, with different messages for people who are untrained in science.
* Mobs that make contact with a bluespace river tile will not instantly vanish, and will instead leave a fading afterimage that disappears completely over 30 seconds. The image itself is just a simple `/obj/effect/temporary` - the mob is deleted immediately. This was inspired by what would theoretically happen if you watched someone fall into a black hole - they'd seem to freeze, then slowly fade to red, then fade out (although in this case it's blue because of space magic.) Items still disappear immediately.
* Touch blueriver tiles with a click now has a 5-second channel. Falling in is still instant.
* Alien walls can now be deconstructed/blown up/whatever, and yield iron ingots.
* Blueriver turfs are no longer a non-spreading subtype of supermatter walls, and instead have their own logic.
* Trimmed typepaths for turfs used in the blueriver by removing a bunch of extraneous base types that aren't actually used anywhere (for instance, `/turf/simulated/floor/away/blueriver/alienfloor` is now just `/turf/simulated/floor/blueriver`.)
* Fixed missing baseturfs for the blueriver areas, which meant they defaulted to space.
* Changed ambience in the blueriver areas, so you no longer have Dwarf Fortress music playing.
  * The underground blueriver is no longer completely silent due to lacking ambience.

## Why It's Good For The Game

I feel that the bluespace river's one of the oldest away sites, and it definitely shows. Gitlens places its original implementation at around 5 years ago, and I don't believe it's received any significant updates since then. I find the concept really neat, and while I don't want to remake it entirely at the moment, I do feel like these changes do some significant good in bringing it up to modern standards. In addition to making it feel more immersive and natural, I like to think it also feels more like it belongs with the rest of the game (due to the grammar and writing standards) and makes use of more modern code to pull off a more fascinating aesthetic and appearance.

Regarding the size of the catwalk: this is, admittedly, probably pretty subjective. It's significantly less dangerous, since you're now pretty much guaranteed to be safe unless someone deliberately unwrenches the railings or tries to push someone into the river. The river is still an instant kill, but it's far from the explorer eater it used to be. I'm up to change this, though, if requested - it just felt more believable and less frustrating for it to be more easily traversable.

This also changes the theming/lore a bit, which I wanted to elucidate further on. I find the concept of "alien X" kind of boring, and the blue river has a lot of "alien" objects that are just slightly recolored/different from normal ones. While I maintained the theming, I also tried to throw in some unpredictability with the scanner and examine texts to hint at a "not fully in reality" thing, but ultimately, I'm leaving it up to individual interpretation. I'm also willing to chat about this, too!

## Did you test it?

Runs and compiles fine. I played several rounds - some with natural spawns, some with admin-placed - to make sure everything works correctly. The open space tiles on the top z-level don't render correctly when admin-spawned, but they still work normally; you just don't see the stairs beneath you. Navigating to the away site works, as does landing on it with the Charon. The river still eats explorers and other people, as well as items, but you can now throw them into it before it does so. Stuff can be dragged out from underground and brought back to the surface.

## Changelog

:cl:
maptweak: The bluespace river away site has received some map adjustments and modernization.
maptweak: An anomaly scanner (Alden-Saraspova counter) can now be used on some objects and turfs in the bluespace river away site for some fluff text.
bugfix: Welding apart floors in the bluespace river's underground z-level no longer creates space tiles.
/:cl:

## Media

dreamseeker_Pz2FgYS3H3

dreamseeker_LzpwrEkU3X

### CODE REVIEW

**Comment by 0sj** [comment]
good pr

**Comment by Cupax3** [comment]
testmerge go brrrrrrrrrrrrrrr

File: maps/away/blueriver/blueriver-2.dmm
Comment (Shadowtail117) [inline maps/away/blueriver/blueriver-2.dmm]: General grammatical fixes.

```suggestion
	info = "Well, it didn't take the three days I was betting on, so I'm out twenty bucks, but we managed to punch through into a cave of some sort with the drilling equipment. We're excavating a ramp now and sending one of the guards down to check that there's no spiders in there or anything. Once they report back, we'll move the whole basecamp down there, and- SO, We've moved down into the cave now, and what we've found is astounding! It looks a LOT like a lake or river of blue goo, but... not? I took some scans but couldn't pick anything up aside from radiation spikes from the pool. I also lowered a swab into the fluid to take a sample, but when I touched the swab to the surface of the... Fluid? The swab just... Ceased And yes, that's the best way to describe it. It just stopped existing in my hand. We figure it's best we DON'T touch the blue stuff for now. one of our help has started moving supplies down into the cave to build a bridge of sorts though across the pool. We spotted some sort of structure on the far side, and the bridge is the only way to get to it, so now we wait."
```

File: maps/away/blueriver/blueriver.dm
Comment (Shadowtail117) [inline maps/away/blueriver/blueriver.dm]: Since the idea (I believe) is that the sensor readout is computer-generated, it should be written less personally:
```suggestion
	desc = "Sensor arrays detect a tiny moon with sub-zero surface conditions and sparse flora, potentially seeded from Earth species.<br><br>Topographical scans highlight the profile of a small vessel located on the western side of a valley surrounded by mineral-rich mountains with no broadcasting transponder. Subterranean energy signatures are present and constantly fluctuating, in excess of expected levels.<br><br>The moon has a very long rotational period, with sunrise in an estimated 56 hours."
```

File: maps/away/blueriver/blueriver.dm
Comment (Shadowtail117) [inline maps/away/blueriver/blueriver.dm]: It might be better to change `generate_planet_name()` instead. Otherwise, this away site will be oddly named compared to all other exoplanets/moons.

File: maps/away/blueriver/blueriver.dm
Comment (Shadowtail117) [inline maps/away/blueriver/blueriver.dm]: "...Urist McGreytide, his body and expression unmoving. They are slowly..." sounds odd because of the sudden change in gender. Here are two suggestions either way, you need the `\the`).

```suggestion
		T.desc = "A picture-perfect image of \the [T], their body and expression unmoving. They are slowly fading away into nothing."
```

```suggestion
		T.desc = "A picture-perfect image of \the [T], \his body and expression unmoving. \He [T.gender == PLURAL ? "are" : "is"] slowly fading away into nothing."
```

**Review by Shadowtail117 — DISMISSED** [review]
Any reason why mobs can't fly over the tiles, since that means they wouldn't be touching it?

File: maps/away/blueriver/blueriver.dm
Comment (Ilysen) [reply maps/away/blueriver/blueriver.dm]: I'll use the gender datums here instead, as that way it'll be cleaner/not specifically have a check for plural.

File: maps/away/blueriver/blueriver.dm
Comment (Ilysen) [reply maps/away/blueriver/blueriver.dm]: This is actually intentional to differentiate it, along with the different rotation on the overmap, since otherwise it's very easy to confuse it for an exoplanet, even though it's much smaller and entirely different. I had a point about this in the OP:

`(which is no longer (planet name), an arctic planetoid, but instead (system name + letter), an arctic moon - for instance, GSC 452 b. This simulates real-world conventions for exoplanetary bodies!)`

Would you prefer I change it regardless?

**Comment by Ilysen** [comment]
The no-flying bit is primarily for consistency, so that mobs can't fly to the edge of the z-level/river and see it abruptly cut off. I could create some invisible walls, but it doesn't feel ideal, and I don't want to break the illusion that this is a continuous anomaly, instead of just a small map. That said, I can change it if you feel it astute - I'd just need to do some map adjustments.

**Comment by Ilysen** [comment]
Commit amended with some changes. I also took the opportunity to rewrite the paper logs and make it more clear of their origin (a handwritten log) and made flying mobs able to traverse the river to a certain distance - they're just blocked by invisible walls past a certain point.

File: maps/away/blueriver/blueriver.dm
Comment (Shadowtail117) [reply maps/away/blueriver/blueriver.dm]: Since there isn't an actual difference between an exoplanet and a moon (they are all just randomly-named and randomly-placed away sites) in terms of game mechanics, I think it's best to keep it standardized. Otherwise it can easily be meta'd that this is *that* away site.

**Comment by Shadowtail117** [comment]
> The no-flying bit is primarily for consistency, so that mobs can't fly to the edge of the z-level/river and see it abruptly cut off. I could create some invisible walls, but it doesn't feel ideal, and I don't want to break the illusion that this is a continuous anomaly, instead of just a small map. That said, I can change it if you feel it astute - I'd just need to do some map adjustments.

Why not simply extend the unbreakable cave walls instead? Unless you make it clear that there is some anomalous gravity effect or the pools of bluespace are ceiling-height or something similar, it might be assumed by a player that they can fly over what is otherwise a simple river.

**Comment by Ilysen** [comment]
That's what I ended up doing, if you check the OP. The river stretches on for a while, but large chunks of it to the south and north have density/can't be crossed. Bumping into these will show a message like "hey, this keeps going for a while, you don't want to get lost by going further."

**Review by Cupax3 — DISMISSED** [review]
It passed the scream test!

File: maps/away/blueriver/blueriver.dm
Comment (Ilysen) [reply maps/away/blueriver/blueriver.dm]: It was previously meta-able by the name of "arctic planetoid" instead of "snow exoplanet". I think we're at a bit of an impasse here, so I'll defer to @Cupax3 here.

Comment (Ilysen) [reply maps/away/blueriver/blueriver.dm]: Quoth cups:
image

**Comment by Ilysen** [comment]
PR amended to give the moon more traditional exoplanet names.

**Review by Shadowtail117 — APPROVED** [review]

**Pull request merged**
